apps/webhook_listeners/lib/Service/TokenService.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\WebhookListeners\Service;
+
+use OC\Authentication\Token\IProvider;
+use OC\Authentication\Token\PublicKeyToken;
+use OCA\WebhookListeners\Db\TemporaryTokenMapper;
+use OCA\WebhookListeners\Db\WebhookListener;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\Authentication\Token\IToken;
+use OCP\IUserManager;
+use OCP\L10N\IFactory;
+use OCP\Security\ISecureRandom;
+
+class TokenService {
+	public function __construct(
+		private IProvider $tokenProvider,
+		private ISecureRandom $random,
+		private TemporaryTokenMapper $tokenMapper,
+		private ITimeFactory $time,
+		private IFactory $l10nFactory,
+		private IUserManager $userManager,
+	) {
+	}
+
+
+	/**
+	 * creates an array which includes two arrays of tokens: 'user_ids' and 'user_roles'
+	 * The array ['user_ids' => ['jane', 'bob'], 'user_roles' => ['owner', 'trigger']]
+	 * as requested tokens in the registered webhook produces a result like
+	 * ['user_ids' => [['jane' => 'abcdtokenabcd1'], ['bob','=> 'abcdtokenabcd2']], 'user_roles' => [['owner' => ['admin' => 'abcdtokenabcd3']], ['trigger' => ['user1' => 'abcdtokenabcd4']]]]
+	 * Created auth tokens are valid for 1 hour.
+	 *
+	 * @param WebhookListener $webhookListener
+	 * @param string|null $triggerUserId the user that triggered the webhook call
+	 * @return array
```

Comment:
```suggestion
	 * @return array{user_ids?:array<string,string>,user_roles?:array{owner?:array<string,string>,trigger?:arrays<string,string>}}
```
Document the array structure please.
Especially with complex structures like this one.
Same should be done for the tokenNeeded array parameter. in other methods.